### PR TITLE
Balanced/corrected firearm and gunsmith repair kits

### PR DESCRIPTION
**Pull request by AbsentlyMultitask**
#### Summary
Bugfixes "Firearm and gunsmith repair kits"

#### Purpose of change

Attempts to address #54056, (relevant #45486). The kits had some properties that were missing and some that were extraneous. This is a quick and dirty fix to the issue referenced. The kits should may be looked at and dealt with like toolboxes, some time down the road.

#### Describe the solution

Added soldering and cauterizing functions to both kits. ~~Removed `CROWBAR` use_action from firearm repair kit, since it's not a component. I think it was an early attempt to address the missing nail prying action from a few builds (months ago?) back.~~ I also added the `HACKSAW` use_action to the gunsmith repair kit. I compared the qualities of both kits, and everything is the same, so I figured the missing hacksaw was an oversight. 

EDIT: replaced `HAMMER` use_action with `CROWBAR` per catdach's suggestion.

#### Describe alternatives you've considered

Ignore the issue and wait for consensus on how these kits should be handled.

#### Testing

Made the changes, spawned the items in-game and check the action menus ~~(duplicate pry action disappeared with removal of crowbar use_action).~~

#### Additional context

I kept the medium battery (GRK) difference from the light battery (FRK), since it makes the GRK a little more "professional" and gameplay attractive, especially with the lower charge cost per use. But I would love any advice on this issue.

### CODE REVIEW

**Comment by catdach** [comment]
The `HAMMER` use_action appears to have been fully integrated into the `CROWBAR` use_action (literally the only thing the `HAMMER` use_action  does is call the `CROWBAR` use_action). It looks like the `HAMMER` action is slated to be removed soon (see the comment in iuse.cpp around line 2384), so `CROWBAR` should probably be used instead.

**Comment by Shodan14** [comment]
Last I tried using the crowbar action also cost 100 charges, not sure if it's fixed.

**Comment by new02DAY** [comment]
> Last I tried using the crowbar action also cost 100 charges, not sure if it's fixed.

#54120 brought that up. That's outside my ability to fix, so it'll unfortunately have to be for another PR.

**Comment by new02DAY** [comment]
Related to #53620. I'm wondering if we should just change the cost per use of the kits to match the soldering iron? Since the soldering iron is the entire basis of the energy costs. The only difference between kits would be battery size. I think it'd be a good compromise until there's a change in how the kits are handled (vs. toolbox changes). Unless there's a reason for the specific charge cost/s for the kits?

**Comment by catdach** [comment]
>Related to #53620. I'm wondering if we should just change the cost per use of the kits to match the soldering iron? Since the soldering iron is the entire basis of the energy costs. The only difference between kits would be battery size. I think it'd be a good compromise until there's a change in how the kits are handled (vs. toolbox changes). Unless there's a reason for the specific charge cost/s for the kits?

I think changing the cost per use might also affect the `GUN_REPAIR` action. (might be worth testing)

**Comment by new02DAY** [comment]
> (might be worth testing)

It definitely does, but I think if anything, it keeps everything in-line with soldering iron per use cost. Or at least, it's acceptable in the short-term.

**Comment by new02DAY** [comment]
Anyone know why this is failing? I have no experience with code.
`(~[slow] ~[.])=> ../tests/iteminfo_test.cpp:2331: FAILED:
(~[slow] ~[.])=>   CHECK( item_info_str( hazmat, repaired ) == "--\n" "<color_c_white>Repair</color> using soldering iron, TEST soldering iron, or extended multitool.\n" )`

**Comment by kevingranade** [comment]
need to bring back iteminfo_test.cpp, that's doing important work.

**Comment by new02DAY** [comment]
> need to bring back iteminfo_test.cpp, that's doing important work.

Woops, thought I was just removing my changes. Do my changes to iteminfo_test.cpp affect the failure I mentioned, or should I remove them? (Sorry, I'm stumbling around blind).

File: tests/iteminfo_test.cpp [hunk 1]
```diff
@@ -2330,7 +2330,7 @@ TEST_CASE( "repairable and with what tools", "[iteminfo][repair]" )
     // FIXME: Use an item that can only be repaired by test tools
     CHECK( item_info_str( hazmat, repaired ) ==
            "--\n"
-           "<color_c_white>Repair</color> using soldering iron, TEST soldering iron, or extended multitool.\n" );
+           "<color_c_white>Repair</color> using soldering iron, TEST soldering iron, firearm repair kit, gunsmith repair kit, or extended multitool.\n" );
```
Comment (kevingranade) [inline tests/iteminfo_test.cpp:2333]: Just the order I think.
```suggestion
           "<color_c_white>Repair</color> using gunsmith repair kit, firearm repair kit, soldering iron, TEST soldering iron, or extended multitool.\n" );
```

Comment (new02DAY) [reply tests/iteminfo_test.cpp]: Thanks for fixing it!

**Pull request merged**